File: csharp-nancyfx/src/Org.OpenAPITools/Models/GetMessages.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools._api_v1.Models
{
    /// <summary>
    /// GetMessages
    /// </summary>
    public sealed class GetMessages:  IEquatable<GetMessages>
    { 
        /// <summary>
        /// AvatarUrl
        /// </summary>
        public Object AvatarUrl { get; private set; }

        /// <summary>
        /// Client
        /// </summary>
        public Object Client { get; private set; }

        /// <summary>
        /// Content
        /// </summary>
        public Object Content { get; private set; }

        /// <summary>
        /// ContentType
        /// </summary>
        public Object ContentType { get; private set; }

        /// <summary>
        /// DisplayRecipient
        /// </summary>
        public Object DisplayRecipient { get; private set; }

        /// <summary>
        /// Id
        /// </summary>
        public Object Id { get; private set; }

        /// <summary>
        /// IsMeMessage
        /// </summary>
        public Object IsMeMessage { get; private set; }

        /// <summary>
        /// Reactions
        /// </summary>
        public Object Reactions { get; private set; }

        /// <summary>
        /// RecipientId
        /// </summary>
        public Object RecipientId { get; private set; }

        /// <summary>
        /// SenderEmail
        /// </summary>
        public Object SenderEmail { get; private set; }

        /// <summary>
        /// SenderFullName
        /// </summary>
        public Object SenderFullName { get; private set; }

        /// <summary>
        /// SenderId
        /// </summary>
        public Object SenderId { get; private set; }

        /// <summary>
        /// SenderRealmStr
        /// </summary>
        public Object SenderRealmStr { get; private set; }

        /// <summary>
        /// StreamId
        /// </summary>
        public Object StreamId { get; private set; }

        /// <summary>
        /// Subject
        /// </summary>
        public Object Subject { get; private set; }

        /// <summary>
        /// TopicLinks
        /// </summary>
        public Object TopicLinks { get; private set; }

        /// <summary>
        /// Submessages
        /// </summary>
        public Object Submessages { get; private set; }

        /// <summary>
        /// Timestamp
        /// </summary>
        public Object Timestamp { get; private set; }

        /// <summary>
        /// Type
        /// </summary>
        public Object Type { get; private set; }

        /// <summary>
        /// The user&#39;s [message flags][message-flags] for the message.  [message-flags]: /api/update-message-flags#available-flags 
        /// </summary>
        public List<string> Flags { get; private set; }

        /// <summary>
        /// The UNIX timestamp for when the message was last edited, in UTC seconds. 
        /// </summary>
        public int? LastEditTimestamp { get; private set; }

        /// <summary>
        /// Only present if keyword search was included among the narrow parameters. HTML content of a queried message that matches the narrow, with &#x60;&lt;span class&#x3D;\&quot;highlight\&quot;&gt;&#x60; elements wrapping the matches for the search keywords. 
        /// </summary>
        public string MatchContent { get; private set; }

        /// <summary>
        /// Only present if keyword search was included among the narrow parameters. HTML-escaped topic of a queried message that matches the narrow, with &#x60;&lt;span class&#x3D;\&quot;highlight\&quot;&gt;&#x60; elements wrapping the matches for the search keywords. 
        /// </summary>
        public string MatchSubject { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use GetMessages.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public GetMessages()
        {
        }

        private GetMessages(Object AvatarUrl, Object Client, Object Content, Object ContentType, Object DisplayRecipient, Object Id, Object IsMeMessage, Object Reactions, Object RecipientId, Object SenderEmail, Object SenderFullName, Object SenderId, Object SenderRealmStr, Object StreamId, Object Subject, Object TopicLinks, Object Submessages, Object Timestamp, Object Type, List<string> Flags, int? LastEditTimestamp, string MatchContent, string MatchSubject)
        {
            
            this.AvatarUrl = AvatarUrl;
            
            this.Client = Client;
            
            this.Content = Content;
            
            this.ContentType = ContentType;
            
            this.DisplayRecipient = DisplayRecipient;
            
            this.Id = Id;
            
            this.IsMeMessage = IsMeMessage;
            
            this.Reactions = Reactions;
            
            this.RecipientId = RecipientId;
            
            this.SenderEmail = SenderEmail;
            
            this.SenderFullName = SenderFullName;
            
            this.SenderId = SenderId;
            
            this.SenderRealmStr = SenderRealmStr;
            
            this.StreamId = StreamId;
            
            this.Subject = Subject;
            
            this.TopicLinks = TopicLinks;
            
            this.Submessages = Submessages;
            
            this.Timestamp = Timestamp;
            
            this.Type = Type;
            
            this.Flags = Flags;
            
            this.LastEditTimestamp = LastEditTimestamp;
            
            this.MatchContent = MatchContent;
            
            this.MatchSubject = MatchSubject;
            
        }

        /// <summary>
        /// Returns builder of GetMessages.
        /// </summary>
        /// <returns>GetMessagesBuilder</returns>
        public static GetMessagesBuilder Builder()
        {
            return new GetMessagesBuilder();
        }

        /// <summary>
        /// Returns GetMessagesBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>GetMessagesBuilder</returns>
        public GetMessagesBuilder With()
        {
            return Builder()
                .AvatarUrl(AvatarUrl)
                .Client(Client)
                .Content(Content)
                .ContentType(ContentType)
                .DisplayRecipient(DisplayRecipient)
                .Id(Id)
                .IsMeMessage(IsMeMessage)
                .Reactions(Reactions)
                .RecipientId(RecipientId)
                .SenderEmail(SenderEmail)
                .SenderFullName(SenderFullName)
                .SenderId(SenderId)
                .SenderRealmStr(SenderRealmStr)
                .StreamId(StreamId)
                .Subject(Subject)
                .TopicLinks(TopicLinks)
                .Submessages(Submessages)
                .Timestamp(Timestamp)
                .Type(Type)
                .Flags(Flags)
                .LastEditTimestamp(LastEditTimestamp)
                .MatchContent(MatchContent)
                .MatchSubject(MatchSubject);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(GetMessages other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (GetMessages.
        /// </summary>
        /// <param name="left">Compared (GetMessages</param>
        /// <param name="right">Compared (GetMessages</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (GetMessages left, GetMessages right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (GetMessages.
        /// </summary>
        /// <param name="left">Compared (GetMessages</param>
        /// <param name="right">Compared (GetMessages</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (GetMessages left, GetMessages right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of GetMessages.
        /// </summary>
        public sealed class GetMessagesBuilder
        {
            private Object _AvatarUrl;
            private Object _Client;
            private Object _Content;
            private Object _ContentType;
            private Object _DisplayRecipient;
            private Object _Id;
            private Object _IsMeMessage;
            private Object _Reactions;
            private Object _RecipientId;
            private Object _SenderEmail;
            private Object _SenderFullName;
            private Object _SenderId;
            private Object _SenderRealmStr;
            private Object _StreamId;
            private Object _Subject;
            private Object _TopicLinks;
            private Object _Submessages;
            private Object _Timestamp;
            private Object _Type;
            private List<string> _Flags;
            private int? _LastEditTimestamp;
            private string _MatchContent;
            private string _MatchSubject;

            internal GetMessagesBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for GetMessages.AvatarUrl property.
            /// </summary>
            /// <param name="value">AvatarUrl</param>
            public GetMessagesBuilder AvatarUrl(Object value)
            {
                _AvatarUrl = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Client property.
            /// </summary>
            /// <param name="value">Client</param>
            public GetMessagesBuilder Client(Object value)
            {
                _Client = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Content property.
            /// </summary>
            /// <param name="value">Content</param>
            public GetMessagesBuilder Content(Object value)
            {
                _Content = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.ContentType property.
            /// </summary>
            /// <param name="value">ContentType</param>
            public GetMessagesBuilder ContentType(Object value)
            {
                _ContentType = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.DisplayRecipient property.
            /// </summary>
            /// <param name="value">DisplayRecipient</param>
            public GetMessagesBuilder DisplayRecipient(Object value)
            {
                _DisplayRecipient = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Id property.
            /// </summary>
            /// <param name="value">Id</param>
            public GetMessagesBuilder Id(Object value)
            {
                _Id = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.IsMeMessage property.
            /// </summary>
            /// <param name="value">IsMeMessage</param>
            public GetMessagesBuilder IsMeMessage(Object value)
            {
                _IsMeMessage = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Reactions property.
            /// </summary>
            /// <param name="value">Reactions</param>
            public GetMessagesBuilder Reactions(Object value)
            {
                _Reactions = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.RecipientId property.
            /// </summary>
            /// <param name="value">RecipientId</param>
            public GetMessagesBuilder RecipientId(Object value)
            {
                _RecipientId = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.SenderEmail property.
            /// </summary>
            /// <param name="value">SenderEmail</param>
            public GetMessagesBuilder SenderEmail(Object value)
            {
                _SenderEmail = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.SenderFullName property.
            /// </summary>
            /// <param name="value">SenderFullName</param>
            public GetMessagesBuilder SenderFullName(Object value)
            {
                _SenderFullName = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.SenderId property.
            /// </summary>
            /// <param name="value">SenderId</param>
            public GetMessagesBuilder SenderId(Object value)
            {
                _SenderId = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.SenderRealmStr property.
            /// </summary>
            /// <param name="value">SenderRealmStr</param>
            public GetMessagesBuilder SenderRealmStr(Object value)
            {
                _SenderRealmStr = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.StreamId property.
            /// </summary>
            /// <param name="value">StreamId</param>
            public GetMessagesBuilder StreamId(Object value)
            {
                _StreamId = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Subject property.
            /// </summary>
            /// <param name="value">Subject</param>
            public GetMessagesBuilder Subject(Object value)
            {
                _Subject = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.TopicLinks property.
            /// </summary>
            /// <param name="value">TopicLinks</param>
            public GetMessagesBuilder TopicLinks(Object value)
            {
                _TopicLinks = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Submessages property.
            /// </summary>
            /// <param name="value">Submessages</param>
            public GetMessagesBuilder Submessages(Object value)
            {
                _Submessages = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Timestamp property.
            /// </summary>
            /// <param name="value">Timestamp</param>
            public GetMessagesBuilder Timestamp(Object value)
            {
                _Timestamp = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Type property.
            /// </summary>
            /// <param name="value">Type</param>
            public GetMessagesBuilder Type(Object value)
            {
                _Type = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.Flags property.
            /// </summary>
            /// <param name="value">The user&#39;s [message flags][message-flags] for the message.  [message-flags]: /api/update-message-flags#available-flags </param>
            public GetMessagesBuilder Flags(List<string> value)
            {
                _Flags = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.LastEditTimestamp property.
            /// </summary>
            /// <param name="value">The UNIX timestamp for when the message was last edited, in UTC seconds. </param>
            public GetMessagesBuilder LastEditTimestamp(int? value)
            {
                _LastEditTimestamp = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.MatchContent property.
            /// </summary>
            /// <param name="value">Only present if keyword search was included among the narrow parameters. HTML content of a queried message that matches the narrow, with &#x60;&lt;span class&#x3D;\&quot;highlight\&quot;&gt;&#x60; elements wrapping the matches for the search keywords. </param>
            public GetMessagesBuilder MatchContent(string value)
            {
                _MatchContent = value;
                return this;
            }

            /// <summary>
            /// Sets value for GetMessages.MatchSubject property.
            /// </summary>
            /// <param name="value">Only present if keyword search was included among the narrow parameters. HTML-escaped topic of a queried message that matches the narrow, with &#x60;&lt;span class&#x3D;\&quot;highlight\&quot;&gt;&#x60; elements wrapping the matches for the search keywords. </param>
            public GetMessagesBuilder MatchSubject(string value)
            {
                _MatchSubject = value;
                return this;
            }


            /// <summary>
            /// Builds instance of GetMessages.
            /// </summary>
            /// <returns>GetMessages</returns>
            public GetMessages Build()
            {
                Validate();
                return new GetMessages(
                    AvatarUrl: _AvatarUrl,
                    Client: _Client,
                    Content: _Content,
                    ContentType: _ContentType,
                    DisplayRecipient: _DisplayRecipient,
                    Id: _Id,
                    IsMeMessage: _IsMeMessage,
                    Reactions: _Reactions,
                    RecipientId: _RecipientId,
                    SenderEmail: _SenderEmail,
                    SenderFullName: _SenderFullName,
                    SenderId: _SenderId,
                    SenderRealmStr: _SenderRealmStr,
                    StreamId: _StreamId,
                    Subject: _Subject,
                    TopicLinks: _TopicLinks,
                    Submessages: _Submessages,
                    Timestamp: _Timestamp,
                    Type: _Type,
                    Flags: _Flags,
                    LastEditTimestamp: _LastEditTimestamp,
                    MatchContent: _MatchContent,
                    MatchSubject: _MatchSubject
                );
            }

            private void Validate()
            { 
            }
        }

        
    }
}

File: java-pkmst/src/main/java/com/prokarma/pkmst/model/GetMessages.java
package com.prokarma.pkmst.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.prokarma.pkmst.model.GetMessagesAllOf;
import com.prokarma.pkmst.model.MessagesBase;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.util.ArrayList;
import java.util.List;
/**
 * Response class to be returned by Api
 * @author pkmst
 *
 */
/**
 * GetMessages
 */

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPKMSTServerCodegen", date = "2021-08-08T21:19:29.098714Z[Etc/UTC]")
public class GetMessages   {
  @JsonProperty("avatar_url")
  private Object avatarUrl = null;

  @JsonProperty("client")
  private Object client = null;

  @JsonProperty("content")
  private Object content = null;

  @JsonProperty("content_type")
  private Object contentType = null;

  @JsonProperty("display_recipient")
  private Object displayRecipient = null;

  @JsonProperty("id")
  private Object id = null;

  @JsonProperty("is_me_message")
  private Object isMeMessage = null;

  @JsonProperty("reactions")
  private Object reactions = null;

  @JsonProperty("recipient_id")
  private Object recipientId = null;

  @JsonProperty("sender_email")
  private Object senderEmail = null;

  @JsonProperty("sender_full_name")
  private Object senderFullName = null;

  @JsonProperty("sender_id")
  private Object senderId = null;

  @JsonProperty("sender_realm_str")
  private Object senderRealmStr = null;

  @JsonProperty("stream_id")
  private Object streamId = null;

  @JsonProperty("subject")
  private Object subject = null;

  @JsonProperty("topic_links")
  private Object topicLinks = null;

  @JsonProperty("submessages")
  private Object submessages = null;

  @JsonProperty("timestamp")
  private Object timestamp = null;

  @JsonProperty("type")
  private Object type = null;

  @JsonProperty("flags")
  
  private List<String> flags = null;

  @JsonProperty("last_edit_timestamp")
  private Integer lastEditTimestamp;

  @JsonProperty("match_content")
  private String matchContent;

  @JsonProperty("match_subject")
  private String matchSubject;

  public GetMessages avatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
    return this;
  }

   /**
   * Get avatarUrl
   * @return avatarUrl
  **/
  @ApiModelProperty(value = "")
  public Object getAvatarUrl() {
    return avatarUrl;
  }

  public void setAvatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
  }

  public GetMessages client(Object client) {
    this.client = client;
    return this;
  }

   /**
   * Get client
   * @return client
  **/
  @ApiModelProperty(value = "")
  public Object getClient() {
    return client;
  }

  public void setClient(Object client) {
    this.client = client;
  }

  public GetMessages content(Object content) {
    this.content = content;
    return this;
  }

   /**
   * Get content
   * @return content
  **/
  @ApiModelProperty(value = "")
  public Object getContent() {
    return content;
  }

  public void setContent(Object content) {
    this.content = content;
  }

  public GetMessages contentType(Object contentType) {
    this.contentType = contentType;
    return this;
  }

   /**
   * Get contentType
   * @return contentType
  **/
  @ApiModelProperty(value = "")
  public Object getContentType() {
    return contentType;
  }

  public void setContentType(Object contentType) {
    this.contentType = contentType;
  }

  public GetMessages displayRecipient(Object displayRecipient) {
    this.displayRecipient = displayRecipient;
    return this;
  }

   /**
   * Get displayRecipient
   * @return displayRecipient
  **/
  @ApiModelProperty(value = "")
  public Object getDisplayRecipient() {
    return displayRecipient;
  }

  public void setDisplayRecipient(Object displayRecipient) {
    this.displayRecipient = displayRecipient;
  }

  public GetMessages id(Object id) {
    this.id = id;
    return this;
  }

   /**
   * Get id
   * @return id
  **/
  @ApiModelProperty(value = "")
  public Object getId() {
    return id;
  }

  public void setId(Object id) {
    this.id = id;
  }

  public GetMessages isMeMessage(Object isMeMessage) {
    this.isMeMessage = isMeMessage;
    return this;
  }

   /**
   * Get isMeMessage
   * @return isMeMessage
  **/
  @ApiModelProperty(value = "")
  public Object getIsMeMessage() {
    return isMeMessage;
  }

  public void setIsMeMessage(Object isMeMessage) {
    this.isMeMessage = isMeMessage;
  }

  public GetMessages reactions(Object reactions) {
    this.reactions = reactions;
    return this;
  }

   /**
   * Get reactions
   * @return reactions
  **/
  @ApiModelProperty(value = "")
  public Object getReactions() {
    return reactions;
  }

  public void setReactions(Object reactions) {
    this.reactions = reactions;
  }

  public GetMessages recipientId(Object recipientId) {
    this.recipientId = recipientId;
    return this;
  }

   /**
   * Get recipientId
   * @return recipientId
  **/
  @ApiModelProperty(value = "")
  public Object getRecipientId() {
    return recipientId;
  }

  public void setRecipientId(Object recipientId) {
    this.recipientId = recipientId;
  }

  public GetMessages senderEmail(Object senderEmail) {
    this.senderEmail = senderEmail;
    return this;
  }

   /**
   * Get senderEmail
   * @return senderEmail
  **/
  @ApiModelProperty(value = "")
  public Object getSenderEmail() {
    return senderEmail;
  }

  public void setSenderEmail(Object senderEmail) {
    this.senderEmail = senderEmail;
  }

  public GetMessages senderFullName(Object senderFullName) {
    this.senderFullName = senderFullName;
    return this;
  }

   /**
   * Get senderFullName
   * @return senderFullName
  **/
  @ApiModelProperty(value = "")
  public Object getSenderFullName() {
    return senderFullName;
  }

  public void setSenderFullName(Object senderFullName) {
    this.senderFullName = senderFullName;
  }

  public GetMessages senderId(Object senderId) {
    this.senderId = senderId;
    return this;
  }

   /**
   * Get senderId
   * @return senderId
  **/
  @ApiModelProperty(value = "")
  public Object getSenderId() {
    return senderId;
  }

  public void setSenderId(Object senderId) {
    this.senderId = senderId;
  }

  public GetMessages senderRealmStr(Object senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
    return this;
  }

   /**
   * Get senderRealmStr
   * @return senderRealmStr
  **/
  @ApiModelProperty(value = "")
  public Object getSenderRealmStr() {
    return senderRealmStr;
  }

  public void setSenderRealmStr(Object senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
  }

  public GetMessages streamId(Object streamId) {
    this.streamId = streamId;
    return this;
  }

   /**
   * Get streamId
   * @return streamId
  **/
  @ApiModelProperty(value = "")
  public Object getStreamId() {
    return streamId;
  }

  public void setStreamId(Object streamId) {
    this.streamId = streamId;
  }

  public GetMessages subject(Object subject) {
    this.subject = subject;
    return this;
  }

   /**
   * Get subject
   * @return subject
  **/
  @ApiModelProperty(value = "")
  public Object getSubject() {
    return subject;
  }

  public void setSubject(Object subject) {
    this.subject = subject;
  }

  public GetMessages topicLinks(Object topicLinks) {
    this.topicLinks = topicLinks;
    return this;
  }

   /**
   * Get topicLinks
   * @return topicLinks
  **/
  @ApiModelProperty(value = "")
  public Object getTopicLinks() {
    return topicLinks;
  }

  public void setTopicLinks(Object topicLinks) {
    this.topicLinks = topicLinks;
  }

  public GetMessages submessages(Object submessages) {
    this.submessages = submessages;
    return this;
  }

   /**
   * Get submessages
   * @return submessages
  **/
  @ApiModelProperty(value = "")
  public Object getSubmessages() {
    return submessages;
  }

  public void setSubmessages(Object submessages) {
    this.submessages = submessages;
  }

  public GetMessages timestamp(Object timestamp) {
    this.timestamp = timestamp;
    return this;
  }

   /**
   * Get timestamp
   * @return timestamp
  **/
  @ApiModelProperty(value = "")
  public Object getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(Object timestamp) {
    this.timestamp = timestamp;
  }

  public GetMessages type(Object type) {
    this.type = type;
    return this;
  }

   /**
   * Get type
   * @return type
  **/
  @ApiModelProperty(value = "")
  public Object getType() {
    return type;
  }

  public void setType(Object type) {
    this.type = type;
  }

  public GetMessages flags(List<String> flags) {
    this.flags = flags;
    return this;
  }

  public GetMessages addFlagsItem(String flagsItem) {
    if (this.flags == null) {
      this.flags = new ArrayList<>();
    }
    this.flags.add(flagsItem);
    return this;
  }

   /**
   * The user's [message flags][message-flags] for the message.  [message-flags]: /api/update-message-flags#available-flags 
   * @return flags
  **/
  @ApiModelProperty(value = "The user's [message flags][message-flags] for the message.  [message-flags]: /api/update-message-flags#available-flags ")
  public List<String> getFlags() {
    return flags;
  }

  public void setFlags(List<String> flags) {
    this.flags = flags;
  }

  public GetMessages lastEditTimestamp(Integer lastEditTimestamp) {
    this.lastEditTimestamp = lastEditTimestamp;
    return this;
  }

   /**
   * The UNIX timestamp for when the message was last edited, in UTC seconds. 
   * @return lastEditTimestamp
  **/
  @ApiModelProperty(value = "The UNIX timestamp for when the message was last edited, in UTC seconds. ")
  public Integer getLastEditTimestamp() {
    return lastEditTimestamp;
  }

  public void setLastEditTimestamp(Integer lastEditTimestamp) {
    this.lastEditTimestamp = lastEditTimestamp;
  }

  public GetMessages matchContent(String matchContent) {
    this.matchContent = matchContent;
    return this;
  }

   /**
   * Only present if keyword search was included among the narrow parameters. HTML content of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. 
   * @return matchContent
  **/
  @ApiModelProperty(value = "Only present if keyword search was included among the narrow parameters. HTML content of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. ")
  public String getMatchContent() {
    return matchContent;
  }

  public void setMatchContent(String matchContent) {
    this.matchContent = matchContent;
  }

  public GetMessages matchSubject(String matchSubject) {
    this.matchSubject = matchSubject;
    return this;
  }

   /**
   * Only present if keyword search was included among the narrow parameters. HTML-escaped topic of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. 
   * @return matchSubject
  **/
  @ApiModelProperty(value = "Only present if keyword search was included among the narrow parameters. HTML-escaped topic of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. ")
  public String getMatchSubject() {
    return matchSubject;
  }

  public void setMatchSubject(String matchSubject) {
    this.matchSubject = matchSubject;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GetMessages getMessages = (GetMessages) o;
    return Objects.equals(this.avatarUrl, getMessages.avatarUrl) &&
        Objects.equals(this.client, getMessages.client) &&
        Objects.equals(this.content, getMessages.content) &&
        Objects.equals(this.contentType, getMessages.contentType) &&
        Objects.equals(this.displayRecipient, getMessages.displayRecipient) &&
        Objects.equals(this.id, getMessages.id) &&
        Objects.equals(this.isMeMessage, getMessages.isMeMessage) &&
        Objects.equals(this.reactions, getMessages.reactions) &&
        Objects.equals(this.recipientId, getMessages.recipientId) &&
        Objects.equals(this.senderEmail, getMessages.senderEmail) &&
        Objects.equals(this.senderFullName, getMessages.senderFullName) &&
        Objects.equals(this.senderId, getMessages.senderId) &&
        Objects.equals(this.senderRealmStr, getMessages.senderRealmStr) &&
        Objects.equals(this.streamId, getMessages.streamId) &&
        Objects.equals(this.subject, getMessages.subject) &&
        Objects.equals(this.topicLinks, getMessages.topicLinks) &&
        Objects.equals(this.submessages, getMessages.submessages) &&
        Objects.equals(this.timestamp, getMessages.timestamp) &&
        Objects.equals(this.type, getMessages.type) &&
        Objects.equals(this.flags, getMessages.flags) &&
        Objects.equals(this.lastEditTimestamp, getMessages.lastEditTimestamp) &&
        Objects.equals(this.matchContent, getMessages.matchContent) &&
        Objects.equals(this.matchSubject, getMessages.matchSubject);
  }

  @Override
  public int hashCode() {
    return Objects.hash(avatarUrl, client, content, contentType, displayRecipient, id, isMeMessage, reactions, recipientId, senderEmail, senderFullName, senderId, senderRealmStr, streamId, subject, topicLinks, submessages, timestamp, type, flags, lastEditTimestamp, matchContent, matchSubject);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GetMessages {\n");
    
    sb.append("    avatarUrl: ").append(toIndentedString(avatarUrl)).append("\n");
    sb.append("    client: ").append(toIndentedString(client)).append("\n");
    sb.append("    content: ").append(toIndentedString(content)).append("\n");
    sb.append("    contentType: ").append(toIndentedString(contentType)).append("\n");
    sb.append("    displayRecipient: ").append(toIndentedString(displayRecipient)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    isMeMessage: ").append(toIndentedString(isMeMessage)).append("\n");
    sb.append("    reactions: ").append(toIndentedString(reactions)).append("\n");
    sb.append("    recipientId: ").append(toIndentedString(recipientId)).append("\n");
    sb.append("    senderEmail: ").append(toIndentedString(senderEmail)).append("\n");
    sb.append("    senderFullName: ").append(toIndentedString(senderFullName)).append("\n");
    sb.append("    senderId: ").append(toIndentedString(senderId)).append("\n");
    sb.append("    senderRealmStr: ").append(toIndentedString(senderRealmStr)).append("\n");
    sb.append("    streamId: ").append(toIndentedString(streamId)).append("\n");
    sb.append("    subject: ").append(toIndentedString(subject)).append("\n");
    sb.append("    topicLinks: ").append(toIndentedString(topicLinks)).append("\n");
    sb.append("    submessages: ").append(toIndentedString(submessages)).append("\n");
    sb.append("    timestamp: ").append(toIndentedString(timestamp)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    flags: ").append(toIndentedString(flags)).append("\n");
    sb.append("    lastEditTimestamp: ").append(toIndentedString(lastEditTimestamp)).append("\n");
    sb.append("    matchContent: ").append(toIndentedString(matchContent)).append("\n");
    sb.append("    matchSubject: ").append(toIndentedString(matchSubject)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

